docker/Dockerfile

```diff
@@ -13,47 +13,129 @@
+    rsync \
+    rsyslog \
+    tar \
+    trickle \
+    unzip \
+    vim \
+    wget \
+    zip \
+    && apt-get remove -y cmake python2  && apt-get autoremove -y
+
+# Cargo
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -o install_script.sh && \
+    sh install_script.sh -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# CMake 3.24
+ARG CMAKE_VERSION=3.24.2
+RUN wget https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.sh \
+    && echo "739d372726cb23129d57a539ce1432453448816e345e1545f6127296926b6754 cmake-${CMAKE_VERSION}-Linux-x86_64.sh" | sha256sum -c \ 
+    && mkdir -p "${HOME}"/.local \
+    && bash ./cmake-${CMAKE_VERSION}-Linux-x86_64.sh --skip-license --prefix="${HOME}"/.local/ \
+    && "${HOME}"/.local/bin/cmake --version \
+    && rm cmake-${CMAKE_VERSION}-Linux-x86_64.sh
+ENV PATH="/root/.local/bin:${PATH}"
+
+# Setup ccache
+RUN ln -s "$(which ccache)" /usr/local/bin/gcc \
+  && ln -s "$(which ccache)" /usr/local/bin/g++ \
+  && ln -s "$(which ccache)" /usr/local/bin/cc \
+  && ln -s "$(which ccache)" /usr/local/bin/c++
+
+# This tag must be equivalent to the hash specified by "builtin-baseline" in vcpkg.json
+ARG VCPKG_COMMIT_OR_TAG=2022.09.27
+ARG VCPKG_ROOT=/vcpkg
 
-# run a copy -L to unfold the symlinkes, and strip all exes
-RUN mkdir -p /scilla/0/bin2/ && cp -L /scilla/0/bin/* /scilla/0/bin2/ && strip /scilla/0/bin2/*
+RUN ccache -p && ccache -z
 
-FROM zilliqa/zilliqa-ccache:${ZILLIQA_CCACHE_VERSION} AS builder
+# If COMMIT_OR_TAG is a branch name or a tag, clone a shallow copy which is
+# faster; if this fails, just clone the full repo and checkout the commit.
+RUN git clone https://github.com/microsoft/vcpkg ${VCPKG_ROOT} \
+  && git -C ${VCPKG_ROOT} checkout ${VCPKG_COMMIT_OR_TAG} \
+  && ${VCPKG_ROOT}/bootstrap-vcpkg.sh
 
-# Manually input tag or commit, can be overwritten by docker build-args
-ARG COMMIT_OR_TAG=v8.4.0
-ARG REPO=https://github.com/Zilliqa/Zilliqa.git
-ARG SOURCE_DIR=/root/zilliqa
-ARG BUILD_DIR=/root/zilliqa/build
-ARG INSTALL_DIR=/usr/local
-ARG VCPKG_ROOT=/vcpkg
+WORKDIR /zilliqa
+
+COPY vcpkg-registry/ vcpkg-registry
+COPY vcpkg-configuration.json .
+COPY vcpkg.json .
+
+RUN ${VCPKG_ROOT}/vcpkg install --triplet=x64-linux-dynamic
+
+COPY src/ src
+COPY cmake/ cmake
+COPY daemon/ daemon
+COPY scripts/ scripts
+COPY constants.xml .
+COPY constants_local.xml .
+COPY build.sh .
+COPY CMakeLists.txt .
+
+ENV INSTALL_DIR=/usr/local
 
 # Build & install Zilliqa
-RUN (( git clone -b ${COMMIT_OR_TAG} --depth 1 ${REPO} ${SOURCE_DIR} || git clone ${REPO} ${SOURCE_DIR} )) \
-  && git -C ${SOURCE_DIR} checkout ${COMMIT_OR_TAG} \
-  && cd ${SOURCE_DIR} && CCACHE_BASEDIR="${SOURCE_DIR}" ./build.sh ninja \
-  && cmake --build "${BUILD_DIR}" --target install \
-  && cd ${SOURCE_DIR}/evm-ds && PATH=${BUILD_DIR}/vcpkg_installed/x64-linux-dynamic/tools/protobuf:$PATH cargo build --release --package evm-ds
+RUN CCACHE_BASEDIR="/zilliqa" ./build.sh ninja \
```

Comment:
Placing the code in /zilliqa used to cause issues with ccache which is why it was moved to /root/zilliqa (so it wouldn't be directly under /).
Might not make much of a difference in this docker file since the previous benefits from ccache are pretty much out the window.

Reply:
Yeah - Good to know thanks, lets see what happens to ccache..